### PR TITLE
Added TabBar padding property

### LINKED ISSUE #80309: TabBar inset padding property
## Use case

Currently there's no way to add inset padding for TabBar widget.

Here's currently available solution vs. proposed solution: 
```
 DefaultTabController(
  length: 6,
  child: Padding(
    padding: EdgeInsets.symmetric(horizontal: 16),
    child: TabBar(
      isScrollable: true,
      tabs: [
        Tab(child: Text("Tab #1")),
        Tab(child: Text("Tab #2")),
        Tab(child: Text("Tab #3")),
        Tab(child: Text("Tab #4")),
        Tab(child: Text("Tab #5")),
        Tab(child: Text("Tab #6")),
      ],
    ),
  ),
),
DefaultTabController(
  length: 6,
  child: TabBar(
    isScrollable: true,
    padding: EdgeInsets.symmetric(horizontal: 16),
    tabs: [
      Tab(child: Text("Tab #1")),
      Tab(child: Text("Tab #2")),
      Tab(child: Text("Tab #3")),
      Tab(child: Text("Tab #4")),
      Tab(child: Text("Tab #5")),
      Tab(child: Text("Tab #6")),
    ],
  ),
),
```

As you can see in this image 2nd TabBar which has inset padding functionality fully fills available width and isn't clipped when scrolled left/right.

## Proposal

Add padding property for TabBar which adds inset for whole widget. #75460

### PR DESCRIPTION
**Pull request by krokyze**
This commit adds padding property to TabBar so that it's possible to nicely inset all tabs in case they're scrollable.

Fixes https://github.com/flutter/flutter/issues/80309

Here's currently available solution vs. using this commits functionality: 
```
 DefaultTabController(
  length: 6,
  child: Padding(
    padding: EdgeInsets.symmetric(horizontal: 16),
    child: TabBar(
      isScrollable: true,
      tabs: [
        Tab(child: Text("Tab #1")),
        Tab(child: Text("Tab #2")),
        Tab(child: Text("Tab #3")),
        Tab(child: Text("Tab #4")),
        Tab(child: Text("Tab #5")),
        Tab(child: Text("Tab #6")),
      ],
    ),
  ),
),
DefaultTabController(
  length: 6,
  child: TabBar(
    isScrollable: true,
    padding: EdgeInsets.symmetric(horizontal: 16),
    tabs: [
      Tab(child: Text("Tab #1")),
      Tab(child: Text("Tab #2")),
      Tab(child: Text("Tab #3")),
      Tab(child: Text("Tab #4")),
      Tab(child: Text("Tab #5")),
      Tab(child: Text("Tab #6")),
    ],
  ),
),
```

As you can see wrapping with `Padding` doesn't give as nice effect as passing the padding to `SingleChildScrollView`.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by Piinks — COMMENTED** [review]
Hi @krokyze It looks like this has not been updated in a while, although I am not sure if this change is necessary. Is there an open issue this PR is solving? It looks like this can already be achieved by using the Padding widget as you point out in the description.

**Comment by krokyze** [comment]
Hey @Piinks the same as ListView has children and supports inset padding TabBar should support it too because it's scrollable. As you can see in my example TabBar wrapped in Padding doesn't take full width and edges are clipped off. This is my open issue/pull request.

Do I need to do anything extra to get this approved?

**Review by Piinks — COMMENTED** [review]
> Hey @Piinks the same as ListView has children and supports inset padding TabBar should support it too because it's scrollable. As you can see in my example TabBar wrapped in Padding doesn't take full width and edges are clipped off. This is my open issue/pull request.
> 
> Do I need to do anything extra to get this approved?

Can you open an issue to reference for this change? It also looks like there are failing tests here. Can you take a look at those? It may just require updating your branch with the latest from master.

**Comment by krokyze** [comment]
@Piinks updated my branch with latest changes from master and created issue #80309

**Review by Piinks — COMMENTED** [review]
Thanks for filing an issue! I am interested to see if we get any feedback from others that would like to see this feature. I am concerned now that there are a lot of different paddings on TabBar, padding, indicatorPadding, labelPadding... this might be a little confusing. Does modifying the other paddings result in your desired effect? I am still not sure why this  is necessary.

**Comment by nt4f04uNd** [comment]
This padding is applied to the scrolling area, and personally I don't see other way to achieve such effect, which makes it necessary

**Comment by krokyze** [comment]
The confusion is probably because labelPadding should had been property of Tab not TabBar, the same as ListView doesn't have childrenPadding but each children can have it's own padding.

Modifying the other paddings can't get desired effect.

File: packages/flutter/lib/src/material/tabs.dart [hunk 1]
```diff
@@ -734,6 +735,9 @@ class TabBar extends StatefulWidget implements PreferredSizeWidget {
   /// share of the available space.
   final bool isScrollable;
 
+  /// The amount of space by which to inset the tab bar.
+  final EdgeInsetsGeometry? padding;
```
Comment (Piinks) [inline packages/flutter/lib/src/material/tabs.dart:739]: Let's document how this behaves when scrollable is true or not. I understand now that when the TabBar is _not_ scrollable, it is no different than just wrapping it in Padding. But when TabBar.scrollable is true then this is necessary to achieve the desired effect of the Padding _being part of the scrollable_. This was a bit nuanced and hard to understand, so we should make it clearer in the docs. :)

Comment (krokyze) [reply packages/flutter/lib/src/material/tabs.dart:739]: Do you have any recommendations? I can change it the way you want it 👍

**Review by Piinks — APPROVED** [review]
LGTM Thank you!

**Pull request merged**